From 04d1192ef680ec798475c3c598b5691345ef1838 Mon Sep 17 00:00:00 2001
From: sunxunou <xunoux.sun@intel.com>
Date: Wed, 27 Mar 2019 15:22:19 +0800
Subject: [PATCH] Optimization check for pointer and array index range

1. Null check for pointer 'g_disk_device' before dereferenced.
2. Check array buf index range.

Change-Id: I17042fab49d8c9c5256fc9cce3baa405a3d847c9
Tracked-On: OAM-78889
Signed-off-by: sunxunou <xunoux.sun@intel.com>
---

diff --git a/libkernelflinger/rpmb/rpmb_nvme.c b/libkernelflinger/rpmb/rpmb_nvme.c
index de84b1c..a7ba331 100644
--- a/libkernelflinger/rpmb/rpmb_nvme.c
+++ b/libkernelflinger/rpmb/rpmb_nvme.c
@@ -534,7 +534,7 @@
 	if (EFI_ERROR(ret))
 		return ret;
 
-	memcpy(buf + (blk_addr & 1) * NVME_RPMB_SECTOR_SIZE / 2, buffer, NVME_RPMB_SECTOR_SIZE);
+	memcpy(buf + (blk_addr & 1) * NVME_RPMB_SECTOR_SIZE / 2, buffer, NVME_RPMB_SECTOR_SIZE / 2);
 	ret = nvme_rpmb_write_data_impl(rpmb_dev, write_counter, 1, blk_addr / 2, buf, key, result);
 
 	return ret;
diff --git a/libkernelflinger/uefi_utils.c b/libkernelflinger/uefi_utils.c
index 651c243..57a49f6 100644
--- a/libkernelflinger/uefi_utils.c
+++ b/libkernelflinger/uefi_utils.c
@@ -514,6 +514,9 @@
 	EFI_HANDLE image;
 	EFI_LOADED_IMAGE *loaded_image;
 
+	if (!part_handle)
+		return EFI_INVALID_PARAMETER;
+
 	edp = FileDevicePath(part_handle, path);
 	if (!edp) {
 		error(L"Couldn't generate a path");
